src/App.jsx

```diff
@@ -14,45 +19,140 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = [...goods];
+  const preparedForReverse = [...preparedGoods];
+  const reversedGoods = preparedForReverse.reverse();
+
+  if (!isReversed) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    reversedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good2.localeCompare(good1);
+
+        case SORT_FIELD_LENGTH:
+          return good2.length - good1.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return reversedGoods;
+  }
```

Comment:
```suggestion
    preparedGoods.sort((good1, good2) => {
      switch (sortField) {
        case SORT_FIELD_NAME:
          return good2.localeCompare(good1);

        case SORT_FIELD_LENGTH:
          return good2.length - good1.length;

        default:
          return 0;
      }
    });

  if (isReversed) {
    return preparedGoods.reverse();
  }
```
just reverse if it needs after sorting.